src/dump.rs

```diff
@@ -0,0 +1,105 @@
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct RustDoc {
+    deps: bool,
+    silence: bool,
+}
+
+impl RustDoc {
+    pub fn new() -> Self {
+        Self {
+            deps: false,
+            silence: false,
+        }
+    }
+
+    /// Include dependencies
+    ///
+    /// Reasons to have this disabled:
+    /// - Faster API extraction
+    /// - Less likely to hit bugs in rustdoc, like
+    ///   - rust-lang/rust#89097
+    ///   - rust-lang/rust#83718
+    ///
+    /// Reasons to have this enabled:
+    /// - Check for accidental inclusion of dependencies in your API
+    /// - Detect breaking changes from dependencies in your API
+    pub fn deps(mut self, yes: bool) -> Self {
+        self.deps = yes;
+        self
+    }
+
+    /// Don't write progress to stderr
+    pub fn silence(mut self, yes: bool) -> Self {
+        self.silence = yes;
+        self
+    }
+
+    pub fn dump(&self, manifest_path: &std::path::Path) -> anyhow::Result<std::path::PathBuf> {
+        let crate_name = crate::manifest::get_package_name(manifest_path)?;
+
+        let metadata = cargo_metadata::MetadataCommand::new()
+            .manifest_path(manifest_path)
+            .no_deps()
+            .exec()?;
+        let manifest_target_directory = metadata
+            .target_directory
+            .as_path()
+            .as_std_path()
+            // HACK: Avoid potential errors when mixing toolchains
+            .join("semver-check/target");
+        let target_dir = manifest_target_directory.as_path();
+
+        let stderr = if self.silence {
+            std::process::Stdio::piped()
+        } else {
+            // Print cargo doc progress
+            std::process::Stdio::inherit()
+        };
+
+        let mut cmd = std::process::Command::new("cargo");
+        cmd.env(
+            "RUSTDOCFLAGS",
+            "-Z unstable-options --document-hidden-items --output-format=json",
+        )
```

Comment:
Interesting! I don't know enough to have an opinion on this, I'm just curious why these are passed through the env instead of in `.args()`

Reply:
At the time, that was how I was used to sending arguments directly to rustdoc through cargo-doc.  I'm assuming it supports `--` like `cargo test` and everything else.

I can go either way on how its implemented in practice.

Reply:
No preference either way. If this works without issues, no need to change it.